control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -501,4 +501,26 @@ impl Registry {
             false => vec![],
         }
     }
+
+    /// Register a frontend node(ex:- csi node) to control plane.
+    pub(crate) async fn register_frontend_node_spec(
```

Comment:
Does the `-` mean anything?

```suggestion
    /// Register a frontend node (ex: a csi node) with the control-plane.
    pub(crate) async fn register_frontend_node(
```

Reply:
frontend_node_spec and frontend_node are two seperate types, that is why I had to put the `spec` to specifically point on to it.

Reply:
hmm here it's not a type, it's an entity, we're registering a frontend node? there's no register for frontend_node type?

Reply:
Right, done.